client/js/chart/index.js

```diff
@@ -0,0 +1,141 @@
+import {SVG_NS} from '../util.js';
+
+/**
+ * Creates an SVG chart inside of `el` with the given data.
+ *
+ * @param {HTMLElement} container The container element for this chart.
+ * @param {Map<string, number[]>} histories The ranking history for each
+ * track.
+ * @param {Date[]} dates The date of each history entry.
+ */
+export function createChart(container, histories, dates) {
+  const svg = document.createElementNS(SVG_NS, 'svg');
+  svg.setAttribute('class', 'chart');
+  svg.append(createDefs());
+
+  const seriesContainer = document.createElementNS(SVG_NS, 'g');
+  seriesContainer.setAttribute('class', 'series-container');
+  svg.append(seriesContainer);
+
+  let index = 0;
+  const colors = ['blue', 'red', 'green', 'purple', 'orange'];
+
+  // get only the first 7 tracks for now
+  for (const history of [...histories.values()].slice(0, 7)) {
+    const color = colors[index % colors.length];
+    const series = createSeries(history, color);
+    seriesContainer.append(series);
+    index++;
+  }
+
+  container.append(svg);
+}
+
+const RUN_SCALE_X = 30;
+const RUN_SCALE_Y = 30;
+
+/**
+ * Creates a new series (set of lines on the chart for a specific song).
+ *
+ * @param {number[]} history The historical positions of this track on the
+ * leaderboard.
+ * @param {string} color The color of this series.
+ * @returns {SVGGElement} A group containing the series.
+ */
+function createSeries(history, color) {
+  const series = document.createElementNS(SVG_NS, 'g');
+  series.setAttribute('class', 'series');
+  series.style.setProperty('--run-color', color);
+
+  let start = 0;
+  let end = 0;
+
+  while (end < history.length) {
+    // go until we find a non-null point
```

Comment:
These comments would be more helpful if they described why we're looking for null and non-null points. Like Nick pointed out, it's important to indicate that null points represent the end of a segment or are otherwise not in a segment. So this particular comment should be something like, "The first non-null history node represents the beginning of a segment."